File: manual/data/java/tests/test_gensim_models_Word2Vec.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_gensim_models_Word2Vec {

    @Test
    public void test_train_word2vec_model() {
        String input = "dog barks;cat meows;dog chases cat";
        function_gensim_models_Word2Vec model = new function_gensim_models_Word2Vec();
        String result = model.train_word2vec_model(input);
        // Assert that the vocabulary contains both "dog" and "cat".
        assertTrue(result.contains("dog") && result.contains("cat"), "Expected result to contain both dog and cat");
    }

    @Test
    public void test_similarity_words_exist() {
        String input = "apple orange;banana grape";
        function_gensim_models_Word2Vec model = new function_gensim_models_Word2Vec();
        String result = model.train_word2vec_model(input);
        // Assert that the vocabulary contains "apple" and "orange".
        assertTrue(result.contains("apple"), "Expected result to contain apple");
        assertTrue(result.contains("orange"), "Expected result to contain orange");
    }

    @Test
    public void test_empty_input() {
        String input = "";
        function_gensim_models_Word2Vec model = new function_gensim_models_Word2Vec();
        String result = model.train_word2vec_model(input);
        assertEquals("Training failed", result, "Expected Training failed for empty input");
    }

    @Test
    public void test_single_word_sentences() {
        String input = "hello;world;test";
        function_gensim_models_Word2Vec model = new function_gensim_models_Word2Vec();
        String result = model.train_word2vec_model(input);
        // Assert that the vocabulary contains each individual word.
        assertTrue(result.contains("hello"), "Expected result to contain hello");
        assertTrue(result.contains("world"), "Expected result to contain world");
        assertTrue(result.contains("test"), "Expected result to contain test");
    }

    @Test
    public void test_large_sentence() {
        // Create a sentence with 100 words: word0 word1 ... word99
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < 100; i++) {
            sb.append("word").append(i);
            if(i != 99) {
                sb.append(" ");
            }
        }
        String sentence = sb.toString();
        function_gensim_models_Word2Vec model = new function_gensim_models_Word2Vec();
        String result = model.train_word2vec_model(sentence);
        // Assert that the vocabulary contains "word0" and "word99".
        assertTrue(result.contains("word0"), "Expected result to contain word0");
        assertTrue(result.contains("word99"), "Expected result to contain word99");
    }
}

File: manual/data/python/tests/test_gensim_models_Word2Vec.py
import unittest
from function_gensim_models_Word2Vec import Word2VecModelTrainer

class TestWord2VecModel(unittest.TestCase):

    def setUp(self):
        self.trainer = Word2VecModelTrainer()

    def test_train_word2vec_model(self):
        result = self.trainer.train_word2vec_model("dog barks;cat meows;dog chases cat")
        self.assertTrue("dog" in result and "cat" in result)

    def test_similarity_words_exist(self):
        result = self.trainer.train_word2vec_model("apple orange;banana grape")
        self.assertIn("apple", result)
        self.assertIn("orange", result)

    def test_empty_input(self):
        result = self.trainer.train_word2vec_model("")
        self.assertEqual(result, "Training failed")

    def test_single_word_sentences(self):
        result = self.trainer.train_word2vec_model("hello;world;test")
        self.assertIn("hello", result)
        self.assertIn("world", result)
        self.assertIn("test", result)

    def test_large_sentence(self):
        sentence = " ".join([f"word{i}" for i in range(100)])
        result = self.trainer.train_word2vec_model(sentence)
        self.assertTrue("word99" in result and "word0" in result)

if __name__ == '__main__':
    unittest.main()
